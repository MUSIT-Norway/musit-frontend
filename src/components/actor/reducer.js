import { Observable } from 'rxjs/Rx';
import actions from "./actions";

const dataFromServer = [
  {
    id: 1,
    fn: 'Jarl parl'
  },
  {
    id: 2,
    fn: 'Line pine'
  },
  {
    id: 3,
    fn: 'Kjell pell'
  },
  {
    id: 4,
    fn: 'Stein sein'
  },
  {
    id: 5,
    fn: 'Knut lut'
  },
  {
    id: 6,
    fn: 'Ellen'
  }
]

const initialState = [];

const ActorReducer$ = Observable.of(() => initialState)
  .merge(
    actions.suggest$.map((update) => () => {
      if (!update) {
        return initialState;
      }
      return dataFromServer.filter(d => {
        const name = d.fn.toLowerCase();
        const query = update.value.toLowerCase();
        return name.indexOf(query) > -1
      })
    }),
    actions.clear$.map(() => () => initialState),
  );

export default Observable.merge(
  ActorReducer$.map(reducer => ["suggest", reducer]),
);