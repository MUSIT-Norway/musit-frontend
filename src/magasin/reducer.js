import { toFrontend } from './mapper';
import * as types from './types';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case types.LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: toFrontend(action.result)
      }
    };
  case types.LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case types.CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case types.DELETE:
    return {
      ...state,
      loading: true
    };
  case types.DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case types.DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;