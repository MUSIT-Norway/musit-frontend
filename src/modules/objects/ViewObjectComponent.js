/* @flow */
import React from 'react';
import {Row, Col, Tabs, Tab, PageHeader, Panel} from 'react-bootstrap';
import type { T_ObjectData } from '../../types/object';
import type { T_Samples } from '../../types/samples';
import type { T_Events } from  '../../types/events';
import EventTableComponent from '../../components/events/eventTableComponent';
import SampleTableComponent from '../../components/samples/sampleTableComponent';


type ViewObjectComponentProps = { objectStore: {objectData: T_ObjectData, events: T_Events, samples: T_Samples} };

export const ViewObjectComponent = ({ objectStore: {objectData, events, samples }}: ViewObjectComponentProps) => {
  console.log(objectData);
  return (
    <div>
      <PageHeader>Objektvisning</PageHeader>
      <Panel>
        <Row>
          <Col md={2}><b>Museumsnr:</b>{' '}{objectData && objectData.museumNo}</Col>
          <Col md={2}><b>Unr:</b>{' '}{objectData && objectData.subNo}</Col>
          <Col md={2}><b>Term:</b>{' '}{objectData && objectData.term}</Col>
        </Row>
      </Panel>
      <Tabs id = "objectDetails">
        <Tab
          title="Hendelser"
          eventKey={0} id="1">
          <EventTableComponent events={events} />
        </Tab>
        <Tab title="Relaterte objekter" eventKey={1} id="2">
          <SampleTableComponent samples={samples}/>
        </Tab>
      </Tabs>
    </div>
  );
};

export default ViewObjectComponent;



