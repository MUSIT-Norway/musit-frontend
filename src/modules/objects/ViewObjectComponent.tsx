// @flow
import React from 'react';
import {
  Tabs,
  Tab,
  PageHeader,
  DropdownButton,
  MenuItem,
  Row,
  Col,
  ControlLabel
} from 'react-bootstrap';
import type { ObjectData } from '../../types/object';
import type { Samples, SampleStatus } from '../../types/samples';
import type { Events } from '../../types/events';
import type { AnalysisTypesObject } from '../../types/analysis';
import type { SampleTypesObject } from '../../types/sample';
import EventTable from '../events/components/EventTable';
import type { PredefinedConservation } from '../../types/predefinedConservation';
import SampleTable from '../sample/components/SampleTable';
import ViewObjectData from './components/ViewObjectData';
import Config from '../../config';
import type { AppSession } from '../../types/appSession';
import type { History } from 'types/Routes';
import { I18n } from 'react-i18nify';

type ViewObjectComponentProps = {
  objectStore: { objectData: ObjectData, events: Events, samples: Samples },
  appSession: AppSession,
  analysisTypes: AnalysisTypesObject,
  sampleTypes: SampleTypesObject,
  predefinedConservation: PredefinedConservation,
  pickObject: Function,
  isItemAdded: Function,
  pickList: Object,
  sampleStatus: SampleStatus,
  history: History,
  loading: boolean
};

export const ViewObjectComponent = ({
  objectStore: { objectData, events, samples },
  appSession,
  analysisTypes,
  sampleTypes,
  predefinedConservation,
  pickObject,
  isItemAdded,
  pickList,
  sampleStatus,
  history,
  loading
}: ViewObjectComponentProps) =>
  !loading ? (
    <div className="container">
      <PageHeader>{I18n.t('musit.objects.objectsView.objectView')}</PageHeader>
      <ViewObjectData objectData={objectData} />

      {appSession.rolesForModules.collectionManagementWrite && (
        <Row>
          <Col md={2} style={{ paddingTop: '5px' }}>
            <ControlLabel>
              {I18n.t('musit.objects.objectsView.createNewEvent')}
            </ControlLabel>
          </Col>
          <Col md={3}>
            <DropdownButton
              title={I18n.t('musit.objects.objectsView.selectEvent')}
              id="bg-nested-dropdown"
            >
              <MenuItem
                eventKey="newAnalysis"
                onClick={() =>
                  history.push({
                    pathname: Config.magasin.urls.client.analysis.addAnalysis(appSession),
                    state: [{ objectData }]
                  })
                }
              >
                {I18n.t('musit.objects.objectsView.newAnalysis')}
              </MenuItem>
              {
                // hide for Archaeology & Ethnography collections
                //appSession &&
                //appSession.collectionId !== '2e4f2455-1b3b-4a04-80a1-ba92715ff613' &&
                //appSession.collectionId !== '88b35138-24b5-4e62-bae4-de80fae7df82' &&
                <MenuItem
                  eventKey="newSample"
                  onClick={() =>
                    history.push({
                      pathname: Config.magasin.urls.client.analysis.addSample(
                        appSession,
                        objectData.uuid
                      )
                    })
                  }
                >
                  {I18n.t('musit.objects.objectsView.newSample')}
                </MenuItem>
              }
              <MenuItem
                eventKey="newConservation"
                onClick={() =>
                  history.push({
                    pathname: Config.magasin.urls.client.conservation.addConservation(
                      appSession
                    ),
                    state: [objectData]
                  })
                }
              >
                {I18n.t('musit.conservation.newConservation')}
              </MenuItem>
            </DropdownButton>
          </Col>
        </Row>
      )}
      <hr />
      <Tabs defaultActiveKey={1} id="events">
        <Tab title={I18n.t('musit.objects.objectsView.events.events')} eventKey={1}>
          <EventTable
            events={events}
            analysisTypes={analysisTypes}
            sampleTypes={sampleTypes}
            conservationTypes={
              predefinedConservation && predefinedConservation.conservationTypes
                ? predefinedConservation.conservationTypes
                : []
            }
            appSession={appSession}
            onClick={event => {
              if (event.type === 'Analysis' || event.type === 'AnalysisCollection') {
                history.push({
                  pathname: Config.magasin.urls.client.analysis.viewAnalysis(
                    appSession,
                    event.id
                  )
                });
              } else if (event.type === 'Conservation') {
                history.push({
                  pathname: Config.magasin.urls.client.conservation.viewConservation(
                    appSession,
                    event.eventId
                  )
                });
              }
            }}
          />
        </Tab>
        <Tab title={I18n.t('musit.objects.objectsView.samples.samples')} eventKey={2}>
          <SampleTable
            samples={samples}
            onClick={sample => {
              history.push({
                pathname: Config.magasin.urls.client.analysis.gotoSample(
                  appSession,
                  sample.objectId
                )
              });
            }}
            pickObject={pickObject}
            isItemAdded={isItemAdded}
            pickList={pickList}
            appSession={appSession}
            objectData={objectData}
            sampleTypes={sampleTypes}
            sampleStatus={sampleStatus}
          />
        </Tab>
      </Tabs>
    </div>
  ) : (
    <div className="loading" />
  );

export default ViewObjectComponent;
