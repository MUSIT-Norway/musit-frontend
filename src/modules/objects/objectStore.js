import { createStore, createAction } from 'react-rxjs/dist/RxStore';
import { Observable } from 'rxjs';
import MusitObject from '../../models/object';
//import Sample from '../../models/sample';
//import Event from '../../models/event';
const initialState = { objectData: {}, events: [], samples: [] };

const loadObjectData =  () => (val) => {
  console.log('Hei', val);
  const object$ = MusitObject.getObjectDetails() (val);
  const events$ = Observable.of([]);
  const samples$ = Observable.of([]);
  const r = Observable.forkJoin(object$,events$, samples$).map(([objectData, events, samples]) => ({ objectData, events, samples }));
  r.map((d)=> console.log('Map', d));
  return r;
};


export const loadObject$  = createAction('loadObject$').switchMap(loadObjectData());
export const clear$ = createAction('clear$');


const reducer$ = (actions) => Observable.merge(
  actions.clear$.map(() => () => initialState),
  actions.loadObject$.map((data) => (state) => ({...state, ...data})));

export const objectStore$  = (actions = { loadObject$,  clear$ }) =>
  createStore('objectStore$', reducer$(actions), Observable.of(initialState));

export default objectStore$();