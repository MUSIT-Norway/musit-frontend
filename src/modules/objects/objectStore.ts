// @flow
import { createStore } from 'react-rxjs';
import { createAction } from '../../shared/react-rxjs-patch';
import { Reducer } from 'react-rxjs';
import { Observable, Subject } from 'rxjs';
import MusitObject from '../../models/object';
import Sample from '../../models/sample';
import Event from '../../models/event';
import { ObjectData } from '../../types/object';
import { simpleGet, simplePost } from '../../shared/RxAjax';
import { Maybe, Star, MUSTFIX, TODO } from '../../types/common';

export type ObjectStoreState = {
  objectData: Maybe<ObjectData>;
  events: Maybe<Array<Star>>;
  samples: Maybe<Array<Star>>;
  loadingObjectData: boolean;
  loadingSamples: boolean;
  loadingEvents: boolean;
};

export const initialState: ObjectStoreState = {
  objectData: null,
  events: [],
  samples: [],
  loadingObjectData: false,
  loadingEvents: false,
  loadingSamples: false
};

type Flag = { [key: string]: boolean };

const setLoading$: Subject<Flag> = new Subject();

const flagLoading = (s: Flag) => () => setLoading$.next(s);

export const loadObject$: Observable<ObjectData> = createAction('loadObject$')
  .do(flagLoading({ loadingObjectData: true }))
  .switchMap(MusitObject.getObjectWithCurrentLocation(simpleGet));

export const loadSampleEvents$: Observable<Star> = createAction('loadSampleEvents$')
  .do(flagLoading({ loadingSamples: true }))
  .switchMap(params => {
    return Sample.loadSampleDataForObject(simpleGet)(params as MUSTFIX).flatMap(
      samples => {
        if (!samples || samples.length === 0) {
          return Observable.of([]);
        }
        return Observable.forkJoin(
          samples.map((sample: TODO) => {
            return MusitObject.getObjectLocation(simpleGet)({
              ...params,
              objectType: 'sample',
              objectId: sample.objectId
            } as MUSTFIX).map(currentLocation => ({ ...sample, currentLocation }));
          })
        );
      }
    );
  });

export const loadMoveAndAnalysisEvents$: Observable<Star> = createAction(
  'loadMoveAndAnalysisEvents$'
)
  .do(flagLoading({ loadingEvents: true }))
  .switchMap(Event.getAnalysesAndMoves(simpleGet, simplePost));

export const clearStore$: Observable<void> = createAction('clearStore$');

type Actions = {
  loadSampleEvents$: Observable<Star>;
  loadMoveAndAnalysisEvents$: Observable<Star>;
  loadObject$: Observable<Star>;
  clearStore$: Observable<Star>;
  setLoading$: Subject<Flag>;
};

const reducer$ = (actions: Actions): Observable<Reducer<ObjectStoreState>> =>
  Observable.merge(
    actions.clearStore$.map(() => () => initialState),
    actions.setLoading$.map(loading => (state: ObjectStoreState) => ({
      ...state,
      ...loading
    })),
    actions.loadObject$.map(objectData => (state: ObjectStoreState) => ({
      ...state,
      objectData,
      loadingObjectData: false
    })),
    actions.loadSampleEvents$.map(samples => (state: ObjectStoreState) => ({
      ...state,
      samples,
      loadingSamples: false
    })),
    actions.loadMoveAndAnalysisEvents$.map(events => (state: ObjectStoreState) => ({
      ...state,
      events,
      loadingEvents: false
    }))
  );

export const store$ = (
  actions: Actions = {
    loadObject$,
    loadSampleEvents$,
    loadMoveAndAnalysisEvents$,
    clearStore$,
    setLoading$
  }
) => createStore('objectStore$', reducer$(actions), initialState);

export default store$();
