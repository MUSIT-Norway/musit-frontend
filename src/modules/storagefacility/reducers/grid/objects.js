import { apiUrl } from '../../../../shared/util';
export const LOAD_SEVERAL = 'musit/storageobject-grid/LOAD_SEVERAL';
export const LOAD_SEVERAL_SUCCESS = 'musit/storageobject-grid/LOAD_SEVERAL_SUCCESS';
export const LOAD_SEVERAL_FAIL = 'musit/storageobject-grid/LOAD_SEVERAL_FAIL';
import Config from '../../../../config';
import MusitObject from '../../../../shared/models/object';

const storageObjectGridReducer = (state = {}, action = {}) => {
  switch (action.type) {
  case LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: {
        ...action.result,
        matches: action.result.matches.map(o => new MusitObject(o))
      }
    };
  case LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      data: []
    };
  default:
    return state;
  }
};

export default storageObjectGridReducer;


export const loadObjects = (id, museumId, collectionId, currentPage, callback) => {
  const url = Config.magasin.urls.thingaggregate.baseUrl;
  return {
    types: [LOAD_SEVERAL, LOAD_SEVERAL_SUCCESS, LOAD_SEVERAL_FAIL],
    promise: (client) => client.get(
      apiUrl(
        `${url(museumId)}/node/${id}/objects?${collectionId.getQuery()}&page=${currentPage || 1}&limit=${Config.magasin.limit}`
      )
    ),
    callback
  };
};
