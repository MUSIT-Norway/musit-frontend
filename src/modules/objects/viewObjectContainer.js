// @flow
import ViewObjectComponent from './ViewObjectComponent';
import { RxInjectLegacy as inject } from 'react-rxjs';
import { emitWarning } from '../../shared/errors';
import objectStore$, {
  loadObject$,
  loadMoveAndAnalysisEvents$,
  loadSampleEvents$,
  clearStore$
} from './objectStore';
import store$, { getAnalysisTypes$ } from '../analysis/analysisStore';
import sampleStore$, { getSampleTypes$ } from '../sample/sampleStore';
import predefinedConservation$, {
  loadConservationTypes$
} from '../../stores/predefinedConservation';
import PropTypes from 'prop-types';
import { Observable } from 'rxjs';
import flowRight from 'lodash/flowRight';
import lifeCycle from '../../shared/lifeCycle';
import { toggleObject$ } from '../../stores/pickList';
import { isItemAdded } from '../../stores/pickList';
import MusitObject from '../../models/object';
import sample from '../../models/sample';
import { simpleGet } from '../../shared/RxAjax';
import { I18n } from 'react-i18nify';

const data: {} = {
  appSession$: { type: PropTypes.instanceOf(Observable).isRequired },
  objectStore$,
  analysisTypes: store$,
  sampleTypes: sampleStore$,
  predefinedConservation: predefinedConservation$,
  pickList$: { type: PropTypes.object.isRequired }
};

const props: {} = props => ({
  ...props,
  loading:
    !(props.sampleTypes.sampleTypes && props.sampleTypes.sampleTypes.find) ||
    props.objectStore.loadingObjectData ||
    props.objectStore.loadingSamples ||
    props.objectStore.loadingEvents,
  pickObject: MusitObject.pickObject(toggleObject$, simpleGet),
  isItemAdded,
  goTo: props.history.push,
  sampleStatus: sample.sampleStatuses
});

const commands: {} = {
  loadObject$,
  loadMoveAndAnalysisEvents$,
  loadSampleEvents$,
  getAnalysisTypes$,
  getSampleTypes$,
  clearStore$,
  loadConservationTypes$
};

export const onMount = ({
  loadObject,
  loadMoveAndAnalysisEvents,
  loadSampleEvents,
  match,
  appSession,
  getAnalysisTypes,
  getSampleTypes,
  loadConservationTypes
}: *) => {
  const uuid: string = match.params.id;
  const museumId: number = appSession.museumId;
  const accessToken: string = appSession.accessToken;
  const collectionId: string = appSession.collectionId;
  const ajaxProps = {
    id: uuid,
    objectId: uuid,
    museumId: museumId,
    token: accessToken,
    collectionId: collectionId
  };
  loadObject(ajaxProps);
  loadSampleEvents(ajaxProps);
  loadMoveAndAnalysisEvents({
    ...ajaxProps,
    callback: {
      onFailure: e => {
        if (e.status === 403 && e.request.url.match('/.*analyses.*/')) {
          emitWarning({
            message: I18n.t('musit.errorMainMessages.objects.notAllowedToSeeAnalysis')
          });
        }
        if (e.status === 403 && e.request.url.match('/.*conservation.*/')) {
          emitWarning({
            message: I18n.t(
              'musit.errorMainMessages.objects.notAllowedToSeeConservations'
            )
          });
        }
        if (e.status === 403 && e.request.url.match('/.*storage.*/')) {
          emitWarning({
            message: I18n.t('musit.errorMainMessages.objects.notAllowedToSeeStorageData')
          });
        }
      }
    }
  });
  getAnalysisTypes(ajaxProps);
  getSampleTypes(ajaxProps);
  loadConservationTypes(ajaxProps);
};

const onUnmount = props => {
  props.clearStore();
};

export default flowRight([
  inject(data, commands, props),
  lifeCycle({ onMount, onUnmount })
])(ViewObjectComponent);
