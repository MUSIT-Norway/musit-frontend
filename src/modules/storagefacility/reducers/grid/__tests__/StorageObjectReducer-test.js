import * as actions from '../objects';
import reducer from '../objects';
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);
import Config from '../../../../../config';
import MuseumId from '../../../../../shared/models/museumId';
import CollectionId from '../../../../../shared/models/collectionId';

const comingFromBackend = {
  totalMatches: 3,
  matches: [
    {
      id: 1,
      identifier: {
        museumNo: 'C666',
        subNo: '34'
      },
      displayName: 'Ã˜ks'
    },
    {
      id: 2,
      identifier: {
        museumNo: 'C666',
        subNo: '31'
      },
      displayName: 'Sverd'
    },
    {
      id: 3,
      identifier: {
        museumNo: 'C666',
        subNo: '38'
      },
      displayName: 'Sommerfugl'
    }
  ]
};

describe('StorageUnitReducer', () => {
  it('creates LOAD_SEVERAL_SUCCESS when fetching data has been done', () => {
    const id = 1;
    const url = `${Config.magasin.urls.thingaggregate.baseUrl(new MuseumId(99))}/node/${id}/objects?collectionIds=ddfggggg&page=1&limit=25`;
    nock('http://localhost')
        .get(url)
        .reply(200, comingFromBackend);
    const store = mockStore();

    return store.dispatch(actions.loadObjects(id, new MuseumId(99), new CollectionId('ddfggggg')))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
  });

  it('no action', () => {
    expect(
        reducer(undefined, undefined)
    ).toMatchSnapshot();
  });

  it('initial action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL
        })
    ).toMatchSnapshot();
  });

  it('success action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_SUCCESS,
          result: {
            totalMatches: 1,
            matches: [
              {
                id: 1,
                term: 'Fugl',
                museumNo: 'CH500',
                subNo: '5'
              }
            ]
          }
        })
    ).toMatchSnapshot();
  });

  it('fail action', () => {
    expect(
        reducer(undefined, {
          type: actions.LOAD_SEVERAL_FAIL,
          error: Error('Some error here.')
        })
    ).toMatchSnapshot();
  });

});
