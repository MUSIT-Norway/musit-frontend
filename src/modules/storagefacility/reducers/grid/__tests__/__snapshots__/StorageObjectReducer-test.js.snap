exports[`StorageUnitReducer creates LOAD_SEVERAL_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageobject-grid/LOAD_SEVERAL"
  },
  Object {
    "result": Object {
      "matches": Array [
        Object {
          "displayName": "Ã˜ks",
          "id": 1,
          "identifier": Object {
            "museumNo": "C666",
            "subNo": "34"
          }
        },
        Object {
          "displayName": "Sverd",
          "id": 2,
          "identifier": Object {
            "museumNo": "C666",
            "subNo": "31"
          }
        },
        Object {
          "displayName": "Sommerfugl",
          "id": 3,
          "identifier": Object {
            "museumNo": "C666",
            "subNo": "38"
          }
        }
      ],
      "totalMatches": 3
    },
    "type": "musit/storageobject-grid/LOAD_SEVERAL_SUCCESS"
  }
]
`;

exports[`StorageUnitReducer fail action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": false
}
`;

exports[`StorageUnitReducer initial action 1`] = `
Object {
  "loading": true
}
`;

exports[`StorageUnitReducer no action 1`] = `Object {}`;

exports[`StorageUnitReducer success action 1`] = `
Object {
  "data": Object {
    "matches": Array [
      MusitObject {
        "breadcrumb": undefined,
        "currentLocationId": undefined,
        "id": 1,
        "mainObjectId": undefined,
        "museumNo": "CH500",
        "path": undefined,
        "pathNames": undefined,
        "subNo": "5",
        "term": "Fugl"
      }
    ],
    "totalMatches": 1
  },
  "loaded": true,
  "loading": false
}
`;
